base_layer/wallet/src/contacts_service/storage/sqlite_db.rs

```diff
@@ -104,13 +104,14 @@ impl ContactsBackend for ContactsServiceSqliteDatabase {
             .map_err(|_| ContactsServiceStorageError::R2d2Error)?;
 
         match op {
-            WriteOperation::Insert(kvp) => match kvp {
-                DbKeyValuePair::Contact(k, c) => {
-                    if let Ok(_) = ContactSql::find(&k.to_vec(), &conn) {
-                        return Err(ContactsServiceStorageError::DuplicateContact);
-                    }
-
-                    ContactSql::from(c).commit(&conn)?;
+            WriteOperation::Upsert(kvp) => match kvp {
+                DbKeyValuePair::Contact(k, c) => match ContactSql::find(&k.to_vec(), &conn) {
```

Comment:
https://sqlite.org/lang_UPSERT.html
might be able to simplify this

Reply:
Diesel does not for Sqlite, it is an open issue: https://github.com/diesel-rs/diesel/issues/1854